practice/pig-latin.py

```diff
@@ -1,58 +1,122 @@
-    elif english_word[0] in CAPITALS_LIST:
-        modified_english_word = english_word.lower()
-        re_capitlaize_word = True
+def flip_punctuation_boolean(word):
+    """Checks for punctuation at end of word and flips a boolean switch if it is present """
+    if word[-1] in PUNCTUATION_LIST:
+        return True
     else:
-        modified_english_word = english_word
-        re_capitlaize_word = False
-
-    if english_word[0] in FIRST_VOWEL:
-        pig_latin_word = modified_english_word + 'yay'
-    elif english_word[0] not in FIRST_VOWEL and english_word[1] in LATER_VOWEL:
-        pig_latin_word = modified_english_word[1:] + modified_english_word[:1] + 'ay'
-    elif english_word[0] not in FIRST_VOWEL and  english_word[1] not in LATER_VOWEL:
-        pig_latin_word = modified_english_word[2:] + modified_english_word[:2] + 'ay'
-
-    if re_capitlaize_word == True:
-         new_phrase += '{} '.format(pig_latin_word.capitalize() + new_punctuation )
+        return False
+
+def save_punctuation(english_word):
+    """saves the punctuation of the english word"""
+    punctuation = english_word[-1]
+    return punctuation
+
+def strip_punctuation(english_word):
+    """cuts punctuation of english word"""
+    return english_word[0:-1]
+
+def convert_english_word_to_pig_latin_word(english_word):
+    """Translates english_word into pig_latin_word considering for up to 4 consonants in a row."""
+    if (english_word[0] in FIRST_VOWEL):
+        pig_latin_word = english_word + 'yay'
+    elif english_word in SPECIAL_CASE:
+        pig_latin_word = english_word + 'yay'
+    elif english_word[1] in LATER_VOWEL:
+        pig_latin_word = english_word[1:] + english_word[:1] + 'ay'
+    elif english_word[2] in LATER_VOWEL:
+        pig_latin_word = english_word[2:] + english_word[:2] + 'ay'
     else:
-            new_phrase += '{} '.format(pig_latin_word + new_punctuation)
-    new_punctuation = ''
-    i +=1
+        pig_latin_word = english_word[3:] + english_word[:3] + 'ay'
+    return pig_latin_word
+
+def capitalize_pig_latin_word(temp_pig_latin_word):
+    """If the english word was capitalized, this function appropriately capitalized the Pig Latin Word"""
+    lowercase_first_letter = temp_pig_latin_word[0]
+    uppercase_first_letter = lowercase_first_letter.capitalize()
+    pig_latin_word = temp_pig_latin_word.replace(lowercase_first_letter, uppercase_first_letter, 1)
+    return pig_latin_word
+
+def paste_punctution_back_on_pig_latin_word (temp_pig_latin_word, punctuation):
+    """Adds punctuation to pig_latin_word as well as adding a space for a nice final phrase."""
+    return temp_pig_latin_word + punctuation
+
+def print_full_pig_latin_phrase (phrase):
+    print(phrase)
+
+
+# 3. Main
 
-# 4. Print
 
-print(new_phrase)
\ No newline at end of file
+def main():
+    """Main Function"""
+    pig_latin_phrase = ''
+    english_phrase = get_phrase_casted_into_list()
+    for english_word in english_phrase:
```

Comment:
Break apart this function further: always solve the item-level problem separately and before the list-level problem and is a specific step as part of the main. Work on the sub problem of converting a single word to Pig Latin as it's own problem first, then the list level, then call that in the main.